packages/backend-app-api/config.d.ts

```diff
@@ -88,6 +88,46 @@ export interface Config {
                */
               subject: string;
             };
+            /**
+             * Restricts what types of access that are permitted for this access
+             * method. If no access restrictions are given, it'll have unlimited
+             * access. This access restriction applies for the framework level;
+             * individual plugins may have their own access control mechanisms
+             * on top of this.
+             */
+            accessRestrictions?: Array<{
```

Comment:
Hmm, wonder if we should avoid adding this to legacy auth? Feels a bit cleaner to add it only once we've introduced new methods for HMAC/asymmetric auth?

Reply:
Sure, we could. Internally it's basically all the same,but we can strip this out

Reply:
Hm thinking about this a bit more, why though? It's not like it's in the old `auth.keys` section. I actually wanted the access restrictions to be _outside of_ the type, in practice, such that you can put it on any external access method no matter what. I'd have wanted to declare this essentially as

``´ts
accessRestrictions?: Array<
   AccessRestriction<'legacy', LegacyOptions>,
   AccessRestriction<'static', StaticOptions>,
>,
```

or the corresponding thing in json-schema, and it just being part of the actual item structure whether you want it or not. But the format isn't particularly amenable to it.

Reply:
Only reason is really to drive adoption of new formats once available. No super strong opinion here though, if you think it's better for symmetry I think we can keep it too.